shm/island.cpp

```diff
@@ -0,0 +1,18 @@
+#include "island.hpp"
+#include "coordinates.hpp"
+
+Island::Island(Coordinates position, Time* time) : position_(position), time_(time) {
+    store_ = new Store(time);
```

Comment:
No raw ptr. Only shared_ptr or unique_ptr.

Reply:
@nauka-programowania-MA  Is this comment regarding whole project or only Store in Island?